models/reviews.models.js

```diff
@@ -57,4 +57,16 @@ exports.fetchReviewComments = (review_id) => {
         }
         return promiseArray[0].rows
     })
+};
+
+exports.insertComment = (review_id, body, author) => {
+    const queryStr2 = `INSERT INTO comments (review_id, body, author) VALUES ($1, $2, $3) RETURNING *`
+
+    return db.query(queryStr2, [review_id, body, author]).then((response) => {
+        return response.rows[0];
+    }).catch((err) => {
+        if (err.code === '23503') {
```

Comment:
I would prefer this was done in an error handler, the error will still reach the error handler with this code so we can still do this logic, but it would be cleaner if we kept error handling in error handlers

Reply:
Also as you are returning into a promise chain, we dont actually need this catch, it will return to the catch of the promise chain in your controller